src/resty/aws/request/execute.lua

```diff
@@ -50,15 +51,22 @@ local function execute_request(signed_request)
                   tostring(err))
   end
 
+  local body, body_reader
 
-  local body do
-    if response.has_body then
-      body, err = response:read_body()
-      if not body then
-        return nil, ("failed reading response body from '%s:%s': %s"):format(
-                      tostring(signed_request.host),
-                      tostring(signed_request.port),
-                      tostring(err))
+  if response.headers["application/vnd.amazon.eventstream"] then
```

Comment:
Should this be `if response.headers["Content-Type"] == "application/vnd.amazon.eventstream" then`?

Reply:
I fixed this locally in my test environment and applied other changes from f0ac80e285b8b31291ac6f12021f64d6d9b81954 , works like a charm in my lambda streaming case.

Reply:
@DiscreteTom I have made changes to the header format in the stream handler, otherwise we are wasting CPU by scanning for "header_key == x"

Can you check your plugin code again?

Minor change, now headers are:

```lua
{
  body = "...",
  headers = {
    key = "value",
    key2 = "value2",
  },
}
```

Reply:
No problem. I updated my code locally against this change and it is ok. Gonna push to my PR when this lib is released.